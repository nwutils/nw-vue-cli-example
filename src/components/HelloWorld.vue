<template>
  <div class="hello">
    <h1 data-test="message">
      {{ msg }}
    </h1>

    <h3 v-if="isDesktop" data-test="versions">
      You are using
      Vue.js (v{{ vueVersion }}),
      NW.js (v{{ versions.nw }}-{{ versions['nw-flavor'] }}),
      Node.js (v{{ versions.node }}),
      and
      Chromium (v{{ versions.chromium }}).
    </h3>
    <h3 v-else data-test="versions">
      You are using Vue.js (v{{ vueVersion }}).
    </h3>

    <div v-if="isDesktop">
      <LinkList
        :links="[{
          name: 'Open this app in a browser to see the page rendered differently.',
          url: 'http://localhost:896' + (devMode ? '4' : '5')
        }]"
      />
    </div>

    <button
      v-if="devMode && isDesktop"
      data-test="toggleDevTools"
      @click="toggleDevTools"
    >
      <template v-if="showDevTools">
        Show
      </template>
      <template v-else>
        Hide
      </template>
      Developer Tools
    </button>

    <p>You can use the resources below to find more information around building your Vue App.</p>
    <LinkList :links="welcomeLinks" />

    <h3>Installed CLI Plugins</h3>
    <LinkList :links="pluginLinks" />

    <h3>Essential Links</h3>
    <LinkList :links="essentialLinks" />

    <h3>Ecosystem</h3>
    <LinkList :links="ecosystemLinks" />

    <template v-if="isDesktop">
      <hr />

      <FsExample />
    </template>
  </div>
</template>

<script>
import Vue from 'vue';

import FsExample from '@/components/FsExample.vue';
import LinkList from '@/components/LinkList.vue';

export default {
  name: 'HelloWorld',
  components: {
    FsExample,
    LinkList
  },
  props: {
    msg: {
      type: String,
      default: 'Welcome to your Vue Desktop App in NW.js!'
    }
  },
  data: function () {
    return {
      showDevTools: true,
      welcomeLinks: [
        {
          name: 'nw-vue-cli-example',
          url: 'https://github.com/nwutils/nw-vue-cli-example'
        },
        {
          name: 'NW.js',
          url: 'https://nwjs.io'
        },
        {
          name: 'nw-vue-devtools',
          url: 'https://github.com/nwutils/nw-vue-devtools'
        },
        {
          name: 'nwjs-builder-phoenix',
          url: 'https://github.com/evshiron/nwjs-builder-phoenix'
        },
        {
          name: 'vue-cli',
          url: 'https://cli.vuejs.org'
        }
      ],
      pluginLinks: [
        {
          name: 'babel',
          url: 'https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-babel'
        },
        {
          name: 'eslint',
          url: 'https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-eslint'
        },
        {
          name: 'unit-jest',
          url: 'https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-unit-jest'
        },
        {
          name: 'e2e-cypress',
          url: 'https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-e2e-cypress'
        }
      ],
      essentialLinks: [
        {
          name: 'Core Docs',
          url: 'https://vuejs.org'
        },
        {
          name: 'Forum',
          url: 'https://forum.vuejs.org'
        },
        {
          name: 'Community Chat',
          url: 'https://chat.vuejs.org'
        },
        {
          name: 'Twitter',
          url: 'https://twitter.com/vuejs'
        },
        {
          name: 'News',
          url: 'https://news.vuejs.org'
        }
      ],
      ecosystemLinks: [
        {
          name: 'vue-router',
          url: 'https://router.vuejs.org'
        },
        {
          name: 'vuex',
          url: 'https://vuex.vuejs.org'
        },
        {
          name: 'vue-devtools',
          url: 'https://github.com/vuejs/vue-devtools#vue-devtools'
        },
        {
          name: 'vue-loader',
          url: 'https://vue-loader.vuejs.org'
        },
        {
          name: 'awesome-vue',
          url: 'https://github.com/vuejs/awesome-vue'
        }
      ]
    };
  },
  methods: {
    toggleDevTools: function () {
      if (this.showDevTools) {
        this.nw.Window.get().showDevTools();
      } else {
        this.nw.Window.get().closeDevTools();
      }
      this.showDevTools = !this.showDevTools;
    }
  },
  computed: {
    devMode: function () {
      return this.isDesktop && this.process.versions['nw-flavor'] === 'sdk';
    },
    versions: function () {
      return this.isDesktop && this.process.versions;
    },
    vueVersion: function () {
      return Vue.version;
    }
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h3 {
  margin: 40px 0px 0px;
}
button {
  background: linear-gradient(0deg, #2EB277, #65E6AC);
  border-width: 3.4px;
  border-color: #35495E;
  border-radius: 8px;
  margin: 22px 0px 6px 0px;
  padding: 11px 17px;
  color: #050709;
  font-family: 'Trebuchet MS', Verdana, sans-serif;
  font-size: 19px;
  font-weight: bold;
}
</style>
