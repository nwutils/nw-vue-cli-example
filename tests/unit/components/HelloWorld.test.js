import HelloWorld from '@/components/HelloWorld.vue';

import testHelpers from '@@/unit/test-helpers.js';

describe('HelloWorld.vue', () => {
  const shared = {
    msg: 'new message',
    renderPropsMsg: function () {
      const wrapper = testHelpers.shallowMount(HelloWorld, {
        props: {
          msg: this.msg
        }
      });

      return wrapper;
    },
    renderDefaultContents: function () {
      const wrapper = testHelpers.mount(HelloWorld);
      return wrapper;
    }
  };

  describe('Desktop', () => {
    test('Render props.msg', () => {
      const wrapper = shared.renderPropsMsg();

      expect(wrapper.find('[data-test="message"]').text())
        .toEqual(shared.msg);
    });

    test('Render default contents', () => {
      const wrapper = shared.renderDefaultContents();

      expect(wrapper)
        .toMatchSnapshot();
    });

    test('Activate dev tools', async () => {
      const wrapper = testHelpers.shallowMount(HelloWorld);

      const button = wrapper.find('[data-test="toggleDevTools"]');

      button.trigger('click');
      await wrapper.vm.$nextTick();

      expect(wrapper.find('[data-test="toggleDevTools"]').html())
        .toMatchSnapshot('hide');

      button.trigger('click');
      await wrapper.vm.$nextTick();

      expect(wrapper.find('[data-test="toggleDevTools"]').html())
        .toMatchSnapshot('show');
    });
  });

  describe('Web', () => {
    beforeEach(() => {
      window.webSetup();
    });

    test('Render props.msg', () => {
      const wrapper = shared.renderPropsMsg();

      expect(wrapper.find('[data-test="message"]').text())
        .toEqual(shared.msg);
    });

    test('Render default contents', () => {
      const wrapper = shared.renderDefaultContents();

      expect(wrapper)
        .toMatchSnapshot();
    });
  });
});
