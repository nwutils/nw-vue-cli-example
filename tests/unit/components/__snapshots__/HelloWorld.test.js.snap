// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue Desktop Activate dev tools: hide 1`] = `
<button>
  Hide

  Developer Tools
</button>
`;

exports[`HelloWorld.vue Desktop Activate dev tools: show 1`] = `
<button>
  Show

  Developer Tools
</button>
`;

exports[`HelloWorld.vue Desktop Render default contents 1`] = `
<div class="hello">
  <h1>
    Welcome to your Vue Desktop App in NW.js!
  </h1>
  <h3>
    You are using
    Vue.js (v2.6.14),
    NW.js (v0.62.2-sdk),
    Node.js (v17.7.2),
    and
    Chromium (v99.0.4844.84).
  </h3>
  <div>
    <ul>
      <li title="http://localhost:8964">
        <a href="#">
          Open this app in a browser to see the page rendered differently.
        </a>
      </li>
    </ul>
  </div>
  <button>
    Show

    Developer Tools
  </button>
  <p>You can use the resources below to find more information around building your Vue App.</p>
  <ul>
    <li title="github.com/nwutils/nw-vue-cli-example">
      <a href="#">
        nw-vue-cli-example
      </a>
    </li>
    <li title="nwjs.io">
      <a href="#">
        NW.js
      </a>
    </li>
    <li title="github.com/nwutils/nw-vue-devtools">
      <a href="#">
        nw-vue-devtools
      </a>
    </li>
    <li title="github.com/evshiron/nwjs-builder-phoenix">
      <a href="#">
        nwjs-builder-phoenix
      </a>
    </li>
    <li title="cli.vuejs.org">
      <a href="#">
        vue-cli
      </a>
    </li>
  </ul>
  <h3>Installed CLI Plugins</h3>
  <ul>
    <li title="github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-babel">
      <a href="#">
        babel
      </a>
    </li>
    <li title="github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-eslint">
      <a href="#">
        eslint
      </a>
    </li>
    <li title="github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-unit-jest">
      <a href="#">
        unit-jest
      </a>
    </li>
    <li title="github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-e2e-cypress">
      <a href="#">
        e2e-cypress
      </a>
    </li>
  </ul>
  <h3>Essential Links</h3>
  <ul>
    <li title="vuejs.org">
      <a href="#">
        Core Docs
      </a>
    </li>
    <li title="forum.vuejs.org">
      <a href="#">
        Forum
      </a>
    </li>
    <li title="chat.vuejs.org">
      <a href="#">
        Community Chat
      </a>
    </li>
    <li title="twitter.com/vuejs">
      <a href="#">
        Twitter
      </a>
    </li>
    <li title="news.vuejs.org">
      <a href="#">
        News
      </a>
    </li>
  </ul>
  <h3>Ecosystem</h3>
  <ul>
    <li title="router.vuejs.org">
      <a href="#">
        vue-router
      </a>
    </li>
    <li title="vuex.vuejs.org">
      <a href="#">
        vuex
      </a>
    </li>
    <li title="github.com/vuejs/vue-devtools#vue-devtools">
      <a href="#">
        vue-devtools
      </a>
    </li>
    <li title="vue-loader.vuejs.org">
      <a href="#">
        vue-loader
      </a>
    </li>
    <li title="github.com/vuejs/awesome-vue">
      <a href="#">
        awesome-vue
      </a>
    </li>
  </ul>
  <hr>
  <div>
    <!---->
    <button>
      Click for File System example
    </button>
    <!---->
  </div>
</div>
`;

exports[`HelloWorld.vue Web Render default contents 1`] = `
<div class="hello">
  <h1>
    Welcome to your Vue Desktop App in NW.js!
  </h1>
  <h3>
    You are using Vue.js (v2.6.14).
  </h3>
  <!---->
  <!---->
  <p>You can use the resources below to find more information around building your Vue App.</p>
  <ul>
    <li title="github.com/nwutils/nw-vue-cli-example">
      <a href="#">
        nw-vue-cli-example
      </a>
    </li>
    <li title="nwjs.io">
      <a href="#">
        NW.js
      </a>
    </li>
    <li title="github.com/nwutils/nw-vue-devtools">
      <a href="#">
        nw-vue-devtools
      </a>
    </li>
    <li title="github.com/evshiron/nwjs-builder-phoenix">
      <a href="#">
        nwjs-builder-phoenix
      </a>
    </li>
    <li title="cli.vuejs.org">
      <a href="#">
        vue-cli
      </a>
    </li>
  </ul>
  <h3>Installed CLI Plugins</h3>
  <ul>
    <li title="github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-babel">
      <a href="#">
        babel
      </a>
    </li>
    <li title="github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-eslint">
      <a href="#">
        eslint
      </a>
    </li>
    <li title="github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-unit-jest">
      <a href="#">
        unit-jest
      </a>
    </li>
    <li title="github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-e2e-cypress">
      <a href="#">
        e2e-cypress
      </a>
    </li>
  </ul>
  <h3>Essential Links</h3>
  <ul>
    <li title="vuejs.org">
      <a href="#">
        Core Docs
      </a>
    </li>
    <li title="forum.vuejs.org">
      <a href="#">
        Forum
      </a>
    </li>
    <li title="chat.vuejs.org">
      <a href="#">
        Community Chat
      </a>
    </li>
    <li title="twitter.com/vuejs">
      <a href="#">
        Twitter
      </a>
    </li>
    <li title="news.vuejs.org">
      <a href="#">
        News
      </a>
    </li>
  </ul>
  <h3>Ecosystem</h3>
  <ul>
    <li title="router.vuejs.org">
      <a href="#">
        vue-router
      </a>
    </li>
    <li title="vuex.vuejs.org">
      <a href="#">
        vuex
      </a>
    </li>
    <li title="github.com/vuejs/vue-devtools#vue-devtools">
      <a href="#">
        vue-devtools
      </a>
    </li>
    <li title="vue-loader.vuejs.org">
      <a href="#">
        vue-loader
      </a>
    </li>
    <li title="github.com/vuejs/awesome-vue">
      <a href="#">
        awesome-vue
      </a>
    </li>
  </ul>
  <!---->
</div>
`;
